tests/actionsFactory.spec.ts

```diff
@@ -58,6 +58,51 @@ describe("actionsFactory test", () => {
         }); 
     });
 
+    describe("create Snippet action", () => {
+        it("suceeds with all the mandatory params", () => {
+            const actionJson = {
+                [ActionJsonKey.ActionType]: ActionType.Snippet,
+                id: "id",
+                snippetName: "name",
+                contributorId: "contributorId",
+                context: {},
+                isNonInteractive: true
+            };
+            const action = ActionsFactory.createAction(actionJson);
+            expect(action instanceof SnippetAction).to.be.true;
+            expect((action as SnippetAction).snippetName).to.be.equal("name");
+            expect((action as SnippetAction).contributorId).to.be.deep.equal("contributorId");
+        });
+
+        it("suceeds with just the mandatory params", () => {
+            const actionJson = {
+                [ActionJsonKey.ActionType]: ActionType.Snippet,
+                snippetName: "name",
+                contributorId: "contributorId"
+            };
+            const action = ActionsFactory.createAction(actionJson);
+            expect(action instanceof SnippetAction).to.be.true;
+            expect((action as SnippetAction).snippetName).to.be.equal("name");
+            expect((action as SnippetAction).contributorId).to.be.deep.equal("contributorId");
+        });
+
+        it("fails without snippetName", () => {
+            const actionJson = {
+                [ActionJsonKey.ActionType]: ActionType.Snippet,
+                contributorId: "contributorId"
+            };
+            expect(() => ActionsFactory.createAction(actionJson)).to.throw(`snippetName is missing for actionType=${ActionType.Snippet}`);    
```

Comment:
You can pass a regexp to `to.throw` so you won't assert on the exact string but instead on some pattern
to make it easier to maintain if/when the error message changes.

```javascript
var badFn = function () { throw new TypeError('Illegal salmon!'); };
expect(badFn).to.throw(/salmon/);
```